docs/proposals/Periodic-Retrieval-of-Certificates-and-Keys-from-AKV.md

```diff
@@ -0,0 +1,121 @@
+
+### Certificate revocation
+
+A malicious user compromised specific versions of certificate. Alice promptly reported this incident to the Certificate Authority (CA). The CA added the compromised versions to the Certificate Revocation List (CRL) and responded to Online Certificate Status Protocol (OCSP) queries with the revoked status. As a result, images signed with the revoked versions fail signature verification. Alice rotated the certificate, obtaining a new version issued by the CA while maintaining the same certificate name and identity information. The build pipeline was triggered to build and sign all the images with the new version of the certificate. Furthermore, as Alice configures Ratify to periodically retrieve all available versions of the certificate from AKV, it ensures successful verification of images signed with the latest version without requiring manual configuration adjustments. By following these actions, Alice maintains security while seamlessly transitioning to the new certificate version.
+
+### Key rotation
+
+Bob, an application engineer at Wabbit Network LLC, focuses on securing containerized applications. He sets up the build pipeline to sign container images using Cosign and key pairs in AKV. To ensure that only trusted images are deployed in Kubernetes clusters, he deploys Gatekeeper and Ratify. Ratify is configured to validate Cosign signatures using keys from AKV. As a security best practice, Bob configured automated key rotation for the key in AKV, resulting a new version of the key created before key expiry. He ensures that the pipeline always uses the latest key version for signing container images. Additionally, Bob configures Ratify to periodically retrieve all available versions of the key from AKV, allowing successful verification of images signed by different key versions without any manual configuration adjustments. Bob knows that images are not allowed to be signed with Cosign using keys that have expired in AKV, but verification using expired key is allowed. He assumes that the old versions of the keys are retrieved by Ratify, because there are still some images around that were signed with old versions.
```

Comment:
`Bob configures Ratify to periodically retrieve all available versions of the key from AKV`. What user behavior does this translate to? Is this a new configuration to introduce (`fetchAllVersion`) or is this saying the user manually added an entry for each version of the key?

Reply:
> `Bob configures Ratify to periodically retrieve all available versions of the key from AKV`. What user behavior does this translate to? Is this a new configuration to introduce (`fetchAllVersion`) or is this saying the user manually added an entry for each version of the key?

I may need to rephase it; to configure periodic retrieve, users just need to configure the key name or certificate name. We may introduce a switch to disable it, then it will be the current default behavior, pull the latest version during configuration, then never update it. But this will be a problem once the key or certificate is rotated.